File: sentry-samples/sentry-samples-spring-boot/src/main/java/io/sentry/samples/spring/boot/CustomEventProcessor.java
package io.sentry.samples.spring.boot;

import io.sentry.EventProcessor;
import io.sentry.SentryEvent;
import io.sentry.protocol.SentryRuntime;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.springframework.boot.SpringBootVersion;
import org.springframework.stereotype.Component;

/**
 * Custom {@link EventProcessor} implementation lets modifying {@link SentryEvent}s before they are
 * sent to Sentry.
 */
@Component
public class CustomEventProcessor implements EventProcessor {
  private final String springBootVersion;

  public CustomEventProcessor(String springBootVersion) {
    this.springBootVersion = springBootVersion;
  }

  public CustomEventProcessor() {
    this(SpringBootVersion.getVersion());
  }

  @Override
  public @NotNull SentryEvent process(@NotNull SentryEvent event, @Nullable Object hint) {
    final SentryRuntime runtime = new SentryRuntime();
    runtime.setVersion(springBootVersion);
    runtime.setName("Spring Boot");
    event.getContexts().setRuntime(runtime);
    return event;
  }
}

File: sentry-samples/sentry-samples-spring-boot-jakarta-kotlin/src/main/kotlin/io/sentry/samples/spring/boot/CustomEventProcessor.kt
package io.sentry.samples.spring.boot

import io.sentry.EventProcessor
import io.sentry.Hint
import io.sentry.SentryEvent
import io.sentry.protocol.SentryRuntime

class CustomEventProcessor(val springBootVersion: String) : EventProcessor {

    override fun process(event: SentryEvent, hint: Hint): SentryEvent {
        return event.also { e ->
            e.contexts.setRuntime(
                SentryRuntime().also { runtime ->
                    runtime.version = springBootVersion
                    runtime.name = "Spring Boot Kotlin"
                }
            )
        }
    }
}
